addons/purchase/models/purchase_order.py

```diff
@@ -39,7 +39,8 @@ def _amount_all(self):
                 amount_untaxed = sum(order_lines.mapped('price_subtotal'))
                 amount_tax = sum(order_lines.mapped('price_tax'))
 
-            order.amount_untaxed = amount_untaxed
+            if order.amount_untaxed != amount_untaxed:
```

Comment:
You can use the `float_compare` function from `float_utils`, or the `compare_amounts` method of `res.currency` which automatically uses the relevant currency's precision for comparison:

```suggestion
            if order.currency_id.compare_amounts(order.amount_untaxed, amount_untaxed):
```